zcash_keys/src/keys.rs

```diff
@@ -652,16 +685,7 @@ impl UnifiedFullViewingKey {
     /// Parses a `UnifiedFullViewingKey` from its [ZIP 316] string encoding.
     ///
     /// [ZIP 316]: https://zips.z.cash/zip-0316
-    pub fn decode<P: consensus::Parameters>(params: &P, encoding: &str) -> Result<Self, String> {
```

Comment:
Since this is an existing API, we probably don't want to entirely remove it; it can remain as a convenience wrapper around the `zcash_address` based approach. We have such convenience methods in a number of places; the address generation methods on this type will be of that sort after this change.

Reply:
I like the idea of the convenience methods. But you _don't_ want them added to the new type?

Reply:
I'm indifferent; if you want them on the new type then adding them there is fine.

Reply:
Great. I like consistency. Folks who switch from UFVK to IFVK for good hygiene ideally don't have to then figure out which functions on those types to switch to.

Reply:
Although, then I would have to replicate the `string` error result if the goal was really consistency, and I feel gross doing that. So maybe I'll leave them off.